### PR TITLE
New serverless pattern - SQS-EventBus using EventBridge Pipes

### PR DESCRIPTION
**Pull request by chaitanyagummadi**
*Issue #, if available:*

https://github.com/aws-samples/serverless-patterns/issues/2467

*Description of changes:*
This pattern establishes an EventBridge Pipe that retrieves messages from an Amazon Simple Queue Service (SQS) queue and sends the events to a Custom EventBus within Amazon EventBridge.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: eventbridge-pipes-sqs-to-eventbus-terraform/README.md [hunk 1]
```diff
@@ -0,0 +1,95 @@
+# Amazon SQS queue to Amazon EventBridge custom event bus using Amazon EventBridge Pipe
+
+This pattern sends SQS messages to an Event Bus using EventBridge Pipe.
```
Comment (bfreiberg) [inline eventbridge-pipes-sqs-to-eventbus-terraform/README.md:3]:
```suggestion
This pattern sends messages from an SQS queue to an EventBridge event bus using an EventBridge pipe.
```

File: eventbridge-pipes-sqs-to-eventbus-terraform/README.md [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+
+
+
```
Comment (bfreiberg) [inline eventbridge-pipes-sqs-to-eventbus-terraform/README.md:6]: delete additional blank lines

File: eventbridge-pipes-sqs-to-eventbus-terraform/README.md [hunk 3]
```diff
@@ -0,0 +1,95 @@
...
+```
+
+Message2:
+```sh
+    aws sqs send-message --queue-url <SQS_url> --message-body '{"orderId":"125a2e1e-d420-482e-8008-5a606f4b2076", "customerId": "a48516db-66aa-4dbc-bb66-a7f058c5ec24", "type": "OLD"}'
+
+```
+
+Optional: You could create an EventBridge rule with the below event pattern on the custom bus and add relevant targets. With this setup, Message2 would be filtered out by the Pipe, and only Message1 would invoke the configured EventBridge target. Ensure to delete the EventBridge rule before the clean up proecess.
```
Comment (bfreiberg) [inline eventbridge-pipes-sqs-to-eventbus-terraform/README.md:65]:
```suggestion
Optional: You could create an EventBridge rule with the below event pattern on the custom bus and add relevant targets. With this setup, Message2 would be filtered out by the Pipe, and only Message1 would invoke the configured EventBridge target. Ensure to delete the EventBridge rule before the clean up process.
```

File: eventbridge-pipes-sqs-to-eventbus-terraform/README.md [hunk 4]
```diff
@@ -0,0 +1,95 @@
...
+
+
```
Comment (bfreiberg) [inline eventbridge-pipes-sqs-to-eventbus-terraform/README.md:67]: delete additional empty lines

File: eventbridge-pipes-sqs-to-eventbus-terraform/README.md [hunk 5]
```diff
@@ -0,0 +1,95 @@
...
+    terraform destroy
+    ```
+3. Enter 'yes' when prompted.
+
+4. Check if all the resources were deleted successfully.
+    ```bash
+    terraform show
+    ```
+----
+Copyright 2023 Amazon.com, Inc. or its affiliates. All Rights Reserved.
```
Comment (bfreiberg) [inline eventbridge-pipes-sqs-to-eventbus-terraform/README.md:93]:
```suggestion
Copyright 2024 Amazon.com, Inc. or its affiliates. All Rights Reserved.
```

File: eventbridge-pipes-sqs-to-eventbus-terraform/example-pattern.json [hunk 6]
```diff
@@ -0,0 +1,59 @@
+{
+  "title": "Amazon SQS queue to Amazon EventBridge custom event bus using Amazon EventBridge Pipes",
```
Comment (bfreiberg) [inline eventbridge-pipes-sqs-to-eventbus-terraform/example-pattern.json:2]: The title can have at most 75 characters, please reword

File: eventbridge-pipes-sqs-to-eventbus-terraform/example-pattern.json [hunk 7]
```diff
@@ -0,0 +1,59 @@
...
+  "description": "This pattern creates an Amazon EventBridge Pipe to deliver messages from an Amazon SQS queue to Amazon EventBridge Event Bus.",
+  "language": "HCL",
```
Comment (bfreiberg) [inline eventbridge-pipes-sqs-to-eventbus-terraform/example-pattern.json:4]:
```suggestion
  "language": "",
```

Comment (bfreiberg) [reply eventbridge-pipes-sqs-to-eventbus-terraform/example-pattern.json:4]: Can be removed as its the default for Terraform anyway

File: eventbridge-pipes-sqs-to-eventbus-terraform/example-pattern.json [hunk 8]
```diff
@@ -0,0 +1,59 @@
...
+  "level": "200",
+  "framework": "terraform",
```
Comment (bfreiberg) [inline eventbridge-pipes-sqs-to-eventbus-terraform/example-pattern.json:6]:
```suggestion
  "framework": "Terraform",
```

File: eventbridge-pipes-sqs-to-eventbus-terraform/example-pattern.json [hunk 9]
```diff
@@ -0,0 +1,59 @@
...
+      {
+        "text": "Amazon EventBridge Pipes architectural patterns blog post",
+        "link": "https://aws.amazon.com/blogs/compute/implementing-architectural-patterns-with-amazon-eventbridge-pipes/"
+      }
+    ]
+  },
+  "deploy": {
+    "text": [
+      "terraform init",
+       "terraform apply"
```
Comment (bfreiberg) [inline eventbridge-pipes-sqs-to-eventbus-terraform/example-pattern.json:38]:
```suggestion
      "terraform init",
      "terraform apply"
```

File: eventbridge-pipes-sqs-to-eventbus-terraform/example-pattern.json [hunk 10]
```diff
@@ -0,0 +1,59 @@
...
+  "cleanup": {
+    "text": [
+      "terraform destroy"
+    ]
+  },
+  "authors": [
+    {
+      "name": "Chaitanya Gummadi",
+      "image": "https://gravatar.com/avatar/a9b30c6e727f613f3f44a44016e36998719404ea7451c750f402a21f2dd72937.jpg?size=256",
+      "bio": "Chaitanya is a Sr. Cloud Support Engineer with Amazon Web Services ( AWS) based in Texas.",
```
Comment (bfreiberg) [inline eventbridge-pipes-sqs-to-eventbus-terraform/example-pattern.json:55]:
```suggestion
      "bio": "Chaitanya is a Senior Cloud Support Engineer with Amazon Web Services (AWS) based in Texas.",
```

File: eventbridge-pipes-sqs-to-eventbus-terraform/main.tf [hunk 11]
```diff
@@ -0,0 +1,115 @@
+terraform {
+  required_providers {
+    aws = {
+      source  = "hashicorp/aws"
+      version = "~> 5.0"
+    }
+  }
+}
+provider "aws" {
```
Comment (bfreiberg) [inline eventbridge-pipes-sqs-to-eventbus-terraform/main.tf:9]: Either explain to the user in the instructions how to deploy to a different region or remove this block

File: eventbridge-webhooks/1-stripe/template.yml
Comment (bfreiberg) [inline eventbridge-webhooks/1-stripe/template.yml:1]: Was this intended to be part of this pattern?

File: eventbridge-webhooks/2-github/template.yaml
Comment (bfreiberg) [inline eventbridge-webhooks/2-github/template.yaml:1]: Was this intended to be part of this pattern?

File: eventbridge-webhooks/3-twilio/template.yaml
Comment (bfreiberg) [inline eventbridge-webhooks/3-twilio/template.yaml:1]: Was this intended to be part of this pattern?

**Review by bfreiberg — CHANGES_REQUESTED** [review]

**Comment by bfreiberg** [comment]
Hi, do you still plan to address the requested changes?

File: eventbridge-pipes-sqs-to-eventbus-terraform/main.tf (hunk 11 above)
Comment (chaitanyagummadi) [reply eventbridge-pipes-sqs-to-eventbus-terraform/main.tf]: Removed the region/profile block

**Comment by chaitanyagummadi** [comment]
> Hi, do you still plan to address the requested changes?

I have implemented the requested changes and squashed the commits to make it cleaner.

File: eventbridge-pipes-sqs-to-eventbus-terraform/example-pattern.json [hunk 12]
```diff
@@ -0,0 +1,59 @@
...
+      "This filtering capability enables you to create precise event processing pipelines, ensuring that only relevant events are propagated to downstream systems.",
+      "Then, other AWS services that are supported as EventBridge targets can consume these events from the EventBus by adding an EventBridge rule with relevant targets."
+    ]
+  },
+  "gitHub": {
+    "template": {
+      "repoURL": "https://github.com/aws-samples/serverless-patterns/tree/main/eventbridge-pipes-sqs-to-eventbus-terraform",
+      "templateURL": "serverless-patterns/eventbridge-pipes-sqs-to-eventbus-terraform",
+      "projectFolder": "eventbridge-pipes-sqs-to-eventbus-terraform",
+      "templateFile": "eventbridge-pipes-sqs-to-eventbus-terraform/main.tf"
```
Comment (bfreiberg) [inline eventbridge-pipes-sqs-to-eventbus-terraform/example-pattern.json:20]:
```suggestion
      "templateFile": "main.tf"
```

File: eventbridge-pipes-sqs-to-eventbus-terraform/example-pattern.json [hunk 13]
```diff
@@ -0,0 +1,59 @@
...
+    ]
+  },
+  "testing": {
+    "text": [
+      "See the README in the GitHub repo for detailed testing instructions."
+    ]
+  },
+  "cleanup": {
+    "text": [
+      "terraform destroy"
```
Comment (bfreiberg) [inline eventbridge-pipes-sqs-to-eventbus-terraform/example-pattern.json:48]:
```suggestion
      "<code>terraform destroy</code>"
```

**Review by bfreiberg — CHANGES_REQUESTED** [review]

**Review by bfreiberg — APPROVED** [review]

**Comment by bfreiberg** [comment]
Looks good, thanks for your contribution. Your PR will be merged to Serverlessland.com soon.

**Pull request merged**